### DISCUSSION TITLE
Using alphine with Print.js

### DISCUSSION THREAD

**Discussion by sqllyw** [alpinejs/alpine#1349] [4. General]
Hi,
I'm using Alphine with print.js, a sample code here:

https://jsfiddle.net/ngfz9tk7/

in the  
````
print_proc() {
                    this.name = "HELLO WORLD"
                    printJS('PRINT', 'html')
                }
````
I assigned the new string to this.name, then printJS, what printed is the old value of this.name, how to use a third party lib like printJS to work with Alpine? Thanks

**Comment by KevinBatdorf** [comment]
You need to give Alpine time to render. Use this instead: `this.$nextTick(() => printJS('PRINT', 'html'))`

**Reply by sqllyw** [reply]
you are superb! thanks
